### PR TITLE
Add annotation for `lhm-shopify` gem

### PR DESCRIPTION
**Pull request by egiurleo**
Shopify uses a forked version of the `lhm` gem. This forked version is slightly different and requires its own annotation. I've copied over the annotation for the `lhm` gem, removed a couple methods that do not appear in the Shopify version, and modified one sig to match the Shopify gem.

## Tophatting 🎩 
I've tested this annotation against Shopify core and it does not cause any type errors.

### CODE REVIEW

File: rbi/annotations/lhm-shopify.rbi [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  def notify(lowest, highest); end
+end
+
+class Lhm::Printer::Dot < Lhm::Printer::Base
+  sig { void }
+  def end; end
+
+  sig { params(_arg0: T.untyped).void }
+  def notify(*_arg0); end
+end
```
Comment (egiurleo) [inline rbi/annotations/lhm-shopify.rbi:90]: I've updated this method signature per [this thread](https://shopify.slack.com/archives/CBXRGT3SB/p1657915200780799).

File: rbi/annotations/lhm-shopify.rbi [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  sig do
+    params(
+      table_name: T.any(String, Symbol),
+      options: T::Hash[Symbol, T.untyped],
+      block: T.proc.params(arg0: Lhm::Migrator).void
+    ).returns(TrueClass)
+  end
+  def change_table(table_name, options = nil, &block); end
+
+  sig { params(run: T.nilable(T::Boolean), options: T.nilable(T::Hash[Symbol, T.untyped])).returns(T::Boolean) }
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:16]:
```suggestion
  sig { params(run: T::Boolean, options: T.nilable(T::Hash[Symbol, T.untyped])).void }
```

File: rbi/annotations/lhm-shopify.rbi [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  sig { returns(Lhm::Connection) }
+  def connection; end
+
+  sig { returns(Logger) }
+  def self.logger; end
+
+  sig { params(new_logger: Logger).returns(Logger) }
+  def self.logger=(new_logger); end
+
+  sig { params(adapter: Lhm::Connection).returns(Lhm::Connection) }
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:28]:
```suggestion
  sig { params(adapter: Lhm::Connection).void }
```

File: rbi/annotations/lhm-shopify.rbi [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+  sig { void }
+  def end; end
+
+  sig { params(_arg0: T.untyped).void }
+  def notify(*_arg0); end
+end
+
+class Lhm::Migrator
+  include Lhm::Command
+  include Lhm::SqlHelper
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:94]: Not needed

File: rbi/annotations/lhm-shopify.rbi [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+  sig { returns(T::Array[String]) }
+  def statements; end
+end
+
+module Lhm::Throttler
+  sig { params(type: T.any(Lhm::Command, Symbol, String, Class), options: T.nilable(T::Hash[T.untyped, T.untyped])).void }
+  def setup_throttler(type, options = nil); end
+
+  sig { returns(T.self_type) }
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:161]: It doesn't look like this property is redefined in subclasses:
```suggestion
  sig { returns(Throttler) }
```

File: rbi/annotations/lhm-shopify.rbi [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+  def throttler; end
+end
+
+class Lhm::Throttler::Time
+  include Lhm::Command
+
+  sig { params(options: T.nilable(T::Hash[T.untyped, T.untyped])).void }
+  def initialize(options = nil); end
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:169]:
```suggestion
  def initialize(options = {}); end
```

File: rbi/annotations/lhm-shopify.rbi [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+  sig { void }
+  def execute; end
+
+  sig { returns(Integer) }
+  def stride; end
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:175]: Same here with `attr_accessor`

File: rbi/annotations/lhm-shopify.rbi [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+class Lhm::Throttler::LegacyTime < Lhm::Throttler::Time
+  sig { params(timeout: T.nilable(Numeric), stride: T.nilable(Integer)).void }
+  def initialize(timeout, stride); end
+end
+
+class Lhm::Throttler::Factory
+  sig do
+    params(
+      type: T.any(Lhm::Command, Symbol, String, Class),
+      options: T.nilable(T::Hash[T.untyped, T.untyped])
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:196]:
```suggestion
      options: T::Hash[T.untyped, T.untyped]
```

File: rbi/annotations/lhm-shopify.rbi [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+    ).returns(T.untyped)
+  end
+  def self.create_throttler(type, options = nil); end
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:199]:
```suggestion
  def self.create_throttler(type, options = {}); end
```

File: rbi/annotations/lhm-shopify.rbi [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+  sig { params(timeout: T.nilable(Numeric), stride: T.nilable(Integer)).void }
+  def initialize(timeout, stride); end
+end
+
+class Lhm::Throttler::Factory
+  sig do
+    params(
+      type: T.any(Lhm::Command, Symbol, String, Class),
+      options: T.nilable(T::Hash[T.untyped, T.untyped])
+    ).returns(T.untyped)
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:197]: Doesn't it return a `Throttler` instance?
```suggestion
    ).returns(Throttler)
```

File: rbi/annotations/lhm-shopify.rbi [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+  def initialize(options = {}); end
+
+  sig { void }
+  def execute; end
+
+  sig { params(arg0: Integer).returns(Integer) }
+  def stride=(arg0); end
+
+  sig { params(arg0: Numeric).returns(Numeric) }
+  def timeout_seconds=(arg0); end
```
Comment (Morriar) [inline rbi/annotations/lhm-shopify.rbi:182]: I think you can change the `attr_reader` into `attr_accessor` and get rid of this: https://github.com/Shopify/lhm/blob/master/lib/lhm/throttler/time.rb#L9-L10

File: rbi/annotations/lhm-shopify.rbi [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+end
+
+class Lhm::Throttler::Time
+  include Lhm::Command
+
+  sig { returns(Integer) }
+  attr_accessor :stride
+
+  sig { returns(Numeric) }
+  attr_accessor :timeout_seconds
```
Comment (egiurleo) [inline rbi/annotations/lhm-shopify.rbi:170]: @Morriar now using `attr_accessor` with the sig for the reader per [Sorbet docs](https://sorbet.org/docs/faq#how-should-i-add-types-to-methods-defined-with-attr_reader) (TIL how to type an `attr_accessor`)

**Pull request merged**